src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/CreateGcsBucketStep.java

```diff
@@ -62,7 +65,15 @@ public StepResult doStep(FlightContext flightContext)
             .build();
 
     StorageCow storageCow = crlService.createStorageCow(projectId);
-    storageCow.create(bucketInfo);
+
+    // Don't try to create it if it already exists. At this point the assumption is
+    // this is a redo and this step created it already.
+    BucketCow existingBucket = storageCow.get(resource.getBucketName());
+    if (existingBucket == null) {
+      storageCow.create(bucketInfo);
+    } else {
+      logger.info("Bucket {} already exists. Continuing.", resource.getBucketName());
```

Comment:
Is it worth cross-checking that the bucket is in the workspace project?
I think the project id of the bucket is retrievable here and the project can be retrieved using the workspaceId.